Improve the Audit Trails views

MikeAlhayek opened the pull request:
Fix #17105

Piedone commented:
These details are already tracked by Audit Trail.

MikeAlhayek commented:
So we have a way to view the user object as we do in content? I am not sure. 

It's useful to that that info in the user object as we do in contentItems

Piedone commented:
This can be useful if the properties are included in `UserIndex` too, so one can filter/order on them. Without that, I don't see the benefit of duplicating these without the rich contextual info that Audit Trail provides.

MikeAlhayek commented:
How do you access the create and modified user using the audit trails? I thought that the Users Audit trail would have revision control, but did not see a way to do it.

Piedone commented:
The last modified, created dates for content items are there for historical reasons (these predate Audit Trail) and because it's useful to query on them. I think it's the same for Users. Why would we just store the same data that's already stored in Audit Trail, and better? And then not make querying available, for which they'd be actually required (for e.g. "list the most recently registered users first" or "list users who registered in November").

Having a separate section for the general Event Info and the specific one below for Audit Trail events is nice, but I don't get why duplicate things for users:
- Created at UTC should be the same as the event timestamp.
- Created by user ID is in the event info.
- The Modified* ones are the same, and we shouldn't display fields with blank values for different events (since there's also a User Updated event, recorded with all the basic details too).

MikeAlhayek commented:
@Piedone okay. I dropped the CreatedUtc, CreatedBy, ModifiedUtc and ModifiedBy. Instead, I updated the trail audit views so they are grouped in such a way someone will understand what the values mean. I think with this format, It's easier to determine who did what.

![image](https://github.com/user-attachments/assets/1aacbe75-e43b-4878-9c65-90370069d255)

![image](https://github.com/user-attachments/assets/7ca1a009-3f9e-4a41-abca-9bf53bf5a3b9)

Piedone commented:
But why not also keep them, just add them to `UserIndex`? I do think with that, it's useful (but not without).

MikeAlhayek commented:
Why would we inflate a index with data that isn't used?

Also, @sebastienros kinda mention in the meeting today that he is also not a fan of these properties as you originally suggested

Piedone commented on src/OrchardCore.Modules/OrchardCore.AuditTrail/Views/AuditTrailEventDetail.DetailAdmin.cshtml:
```diff
@@ -1,29 +1,31 @@
 @model AuditTrailEventViewModel
 
-<table class="table table-sm table-hover mb-0">
+<h3>@T["Event Info"]</h3>
```
What's above this on the page is `h1`, so this should be an `h2`. Same for Content Item Info and User Info.

MikeAlhayek replied:
it's makes the font way too large.

MikeAlhayek replied:
I changed to to `<h2 class="h3"></h2>`

Piedone replied:
That's what a Hungarian saying calls "iron hoop made of wood" :D.

Piedone reviewed (changes requested):
Because querying on users by these is a use case I can imagine being useful. But not insisting on it if we throw out the properties anyway.
